src/MissingUpdate.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * MissingUpdate.js
+ *
+ * Representation of a missing update
+ */
+import _ from "lodash";
+const updateOutlier = 60000; // 60 seconds
+import Stats from "./Stats";
+import Utils from "./Utils";
+let computedOutlier = 0;
+
+class MissingUpdate {
+  constructor(idx, prevEntry, curEntry) {
+    const interval = curEntry.date - prevEntry.date;
+    const curLoc = _.get(curEntry, "jsonPayload.request.vehicle.lastLocation");
+    const prevLoc = _.get(
+      prevEntry,
+      "jsonPayload.request.vehicle.lastLocation"
+    );
+    const startLoc = new google.maps.LatLng({
+      lat: prevLoc.rawLocation.latitude,
+      lng: prevLoc.rawLocation.longitude,
+    });
+    const endLoc = new google.maps.LatLng({
+      lat: curLoc.rawLocation.latitude,
+      lng: curLoc.rawLocation.longitude,
+    });
+    this.entry = curEntry;
+    this.prevEntry = prevEntry;
+    this.interval = interval;
+    this.startLoc = startLoc;
+    this.startDate = prevEntry.date;
+    this.endDate = curEntry.date;
+    this.endLoc = endLoc;
+    this.idx = idx;
+    this.startVehicleState = _.get(curEntry, "jsonPayload.response.state");
+    this.endVehicleState = _.get(prevEntry, "jsonPayload.response.state");
+    this.duration = Utils.formatDuration(this.interval);
+  }
+
+  /*
+   * Returns data about the update to show in json viewer
+   */
+  getFeaturedData() {
+    return {
+      duration: this.duration,
+      interval: this.interval,
+      startDate: this.startDate,
+      startLoc: this.startLoc.toString(),
+      endDate: this.endDate,
+      endLoc: this.endLoc.toString(),
+      startVehicleState: this.startVehicleState,
+      endVehicleState: this.endVehicleState,
+      computedOutlier: Utils.formatDuration(computedOutlier),
+    };
+  }
+
+  /*
+   * format a vehicle state transitino into something a
+   * human can easily read.
+   */
+  getStateTransition() {
+    const start = this.startVehicleState.replace("VEHICLE_STATE_", "");
```

Comment:
Is this meant to work for delivery vehicles, as well?

Reply:
Eventually yes.    This will still work, just be ugly.  There's going to be a fair amount of refactoring once I really dive into LMFS logs (there's other places where I "shorten" enums so they fit better into columns).